mavlink_vehicles/mavlink_vehicles.cc

```diff
@@ -170,6 +171,30 @@ double dist(global_pos_int p1, global_pos_int p2)
     local_pos res = global_to_local_ned(p1, p2);
     return sqrt(res.x * res.x + res.y * res.y + res.z * res.z);
 }
+
+double get_waypoint_rel_angle(global_pos_int wp_pos, global_pos_int ref_pos,
+                              attitude ref_att)
+{
+    local_pos wp_rel = global_to_local_ned(wp_pos, ref_pos);
+
+    // Calculate relative angle
+    double wp_angle = math::rad2deg(atan2(wp_rel.y, wp_rel.x));
+
+    // While yaw follow NED convention, wp_angle is calculated according to ENU
+    // conventions. We need to follow the same convention before subracting them
+    // up to calculate the relative angle.
+    wp_angle = -wp_angle + 90.0;
+
+    double wp_rel_angle = rad2deg(ref_att.yaw) - wp_angle;
+
+    if (wp_rel_angle > 180.0) {
+        wp_rel_angle = wp_rel_angle - 360.0;
+    } else if (wp_rel_angle < -180.0) {
+        wp_rel_angle = wp_rel_angle + 360.0;
+    }
+
+    return wp_rel_angle;
+}
```

Comment:
The return value of this function will be used a parameter for a subsequent call to rotate(double angle_deg) in [here](https://github.com/01org/mavlink-vehicles/pull/5/files#diff-2801e8d06748835400032a671da7d476R1257). Since rotate() receives and angle in degrees, I decided to have get_waypoint_rel_angle to return in degrees as well. However, the decision to use degrees in rotate() was completely arbitrary. Rethinking about it, I agree that using radians instead, both in rotate() and get_waypoint_rel_angle() would avoid lots of conversions.